### DISCUSSION TITLE
How can I use FCM when I have this module installed?

### DISCUSSION THREAD

**Discussion by FamilyMan** [vuejs/vuefire#1400] [Questions and Help] [answered]
I am using this in Nuxt 3 (vuefire along with nuxt-vuefire). I can't find any documentation or examples of how I can use FCM with this configuration. Does anybody have any pointers for me?
Thanks!

**Comment by posva** [comment, marked as answer]
You can retrieve the firebase app with `useFirebaseApp()` in any `<script setup>` or in plugins, in nuxt. You can then pass that to any service, including FCM:

```vue
<script setup>
import { useFirebaseApp } from 'vuefire'
import { getMessaging } from 'firebase/messaging'

const firebaseApp = useFirebaseApp()
const messaging = getMessaging(firebaseApp)
</script>
```

I think I should add a page to docs as this applies to pretty much any Firebase service, it's not in VueFire's goals to provide wrappers for everything

**Reply by FamilyMan** [reply]
Thank you!
